src/command.c

```diff
@@ -679,8 +679,9 @@ static bool cmd_heapinfo(target_s *t, int argc, const char **argv)
 		target_addr_t heap_limit = strtoul(argv[2], NULL, 16);
 		target_addr_t stack_base = strtoul(argv[3], NULL, 16);
 		target_addr_t stack_limit = strtoul(argv[4], NULL, 16);
-		gdb_outf("heapinfo heap_base: %p heap_limit: %p stack_base: %p stack_limit: %p\n", heap_base, heap_limit,
-			stack_base, stack_limit);
+		gdb_outf("heapinfo heap_base: 0x%08" PRIx32 " heap_limit: 0x%08" PRIx32 " stack_base: 0x%08" PRIx32
```

Comment:
To save a tiny bit of space here, and because the radix is quite obvious we think.. perhaps drop the `0x` suffix? We'd be interested in your thoughts there.

Reply:
I don't use it, this is Koen's code, and it comes without help/usage in-tree, but out-of-tree examples suggest `monitor heapinfo 0x1 0x2 0x3 0x4` to which BMD answers similarly. There is no easy way to read back existing heapinfo values because of encapsulation. %p corresponded to 0x%#x (lx).

Reply:
Right, fair - just think dropping the `0x`'s in the output could be more elegant here while limiting the firmware bloat; but this works as is, so

Reply:
If any software consumers of BMD remote monitor messages exist and use `scanf()` to parse the result, then `man 3 scanf` states that it should consume %p, %x and %X similarly, and discard the optional `0x` or `0X` prefix. I haven't noticed anything like this in the desktop software stack from compuphase. Not sure about Eclipse and other IDEs / graphical debugging frontends with semihosting support.
Wrt people reading the results I suggest just inviting @koendv for review, if you're after these 20 bytes of BMP flash space (concise lowercase %p versus equivalent 0x%08lx as PRIx32, versus prefix-less %x or %X, four times). Excluding the duplicate 54-byte usage message and referring to monitor help is more effective.

Reply:
We would certainly hope people aren't screen scraping monitor output like that - if we worried about that, we'd never be able to change or fix messages produced by the project.

That said, we cannot use %p here because it is not defined for the conversion in question - converting an integer - only for pointers. See the [cppreference page on fprintf()](https://en.cppreference.com/w/c/io/fprintf) for more there. The compiller should (with gdb_outf() defined as having a format string for its first argument) be warning on this.

We'll table trying to claw back a little space for the moment though with just taking the like-output-for-like-output change and we can try and improve the Flash usage of this command at a later date.